### PR TITLE
nodePackages.autoprefixer: Migrate from node-packages

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes
Moves to pkgs/by-name. No update to the actual package. There was a single update that I had to make to postcss-cli to allow it to build, and I'm not sure what exactly caused it, but fixing that hash allows it to build properly.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by fpletz — APPROVED** [review]

File: pkgs/by-name/au/autoprefixer/package.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    pnpm_9.configHook
+  ];
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-sGcqM87xR9XTL/MUO7fGpI1cPK7EgJNpeYwBmqVNB6I=";
+  };
+
+  installPhase = ''
+    mkdir $out
```
Comment (natsukium) [inline pkgs/by-name/au/autoprefixer/package.nix:31]:
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir $out
```

File: pkgs/by-name/au/autoprefixer/package.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    mv bin/ $out
+    mv lib/ $out
+    mv node_modules/ $out
+    mv data/ $out
+    mv package.json $out
+  '';
```
Comment (natsukium) [inline pkgs/by-name/au/autoprefixer/package.nix:37]:
```suggestion
    mv package.json $out

    runHook postInstall
  '';
```
Please add runHook if you override the default phases.

File: pkgs/by-name/au/autoprefixer/package.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  fixupPhase = ''
```
Comment (natsukium) [inline pkgs/by-name/au/autoprefixer/package.nix:39]:
```suggestion
  postFixup = ''
```

Comment (pyrox0) [reply pkgs/by-name/au/autoprefixer/package.nix:39]: Fixed

**Review by Luflosi — APPROVED** [review]

**Comment by trofi** [comment]
The move broke the eval of `autoprefixer.tests` as `packageName` is not a think anymore. Proposed the fix as:
- https://github.com/NixOS/nixpkgs/pull/329253

**Comment by Luflosi** [comment]
@pyrox0 you checked the "Tested, as applicable" box. Which tests did you execute?

**Comment by pyrox0** [comment]
> @pyrox0 you checked the "Tested, as applicable" box. Which tests did you execute?

I had run the nixosTests for the package. I'm not sure why they worked for me, it may have been a caching error or me forgetting to commit.

**Pull request merged**